psydac/api/essential_bc.py

```diff
@@ -43,6 +43,28 @@ def check_boundary_type(bc):
 
 #==============================================================================
 def apply_essential_bc_stencil(a, *, axis, ext, order, identity=False):
+    """ This function applies the homogeneous boundary condition to the Stencil objects,
+        by setting the boundary degrees of freedom to zero in the StencilVector,
+        and the corresponding rows in the StencilMatrix/StencilInterfaceMatrix to zeros.
+        If the identity keyword argument is set to True, the boundary diagonal terms are set to 1.
+
+    Parameters
+    ----------
+    a : LinearOperator/Vector
+        The LinearOperator/Vector that will be modified.
```

Comment:
Thanks for adding this nice docstring! I think we need to be a bit more specific here: the only types that we support are `StencilVector`, `StencilMatrix` and `StencilInterfaceMatrix`.

Reply:
done